### DISCUSSION TITLE
[Feature Idea] Move to using libraries instead of import

### DISCUSSION THREAD

**Discussion by ddshd** [simulot/immich-go#38] [Ideas]
It seems to be that immich is likely going to get rid of the import feature in the future in exchange for the libraries feature newly announced. May be worth looking into using the immich api to create library (or choosing an existing one), adding a path to the processed images, calling a scan.

Maybe we can ask Immich devs to add any missing API endpoints to do the functions above

**Comment by simulot** [comment]
What would be the purpose of this feature?

**Reply by ddshd** [reply]
To allow importing of Google photos library into Immich instead of uploading. Since the import feature of Immich is going away in exchange for Immich libraries it may be worth switching to support that instead.

**Reply by simulot** [reply]
The layout of takeout archive ins't good at all. 

The folder layout isn't good: 1 folder per year, and crazy naming to avoid duplicates names.
Files are duplicates when included in albums.

It means that immich-go would have to reorganize the takeout file to remove unnecessary files and json files, not used by Immich.
I'd rather not go to that direction. This job is well done by the immich server. Images are placed in a tree per year and day.

Anyway, could you show me where it's explained that the import feature will be removed?

**Reply by ddshd** [reply]
Maybe we're confused here, I'm talked about the new libraries feature (not the `upload/library` folder): https://documentation.immich.app/docs/features/libraries#external-libraries.

With this you just can point immich to a folder location and it will process all the files in the directory and will automatically check the directory for new updates. This directory for example could be the result from immich-go's processing of the Google photos takeout folder (images and videos with the proper metadata embedded).

**Reply by ddshd** [reply]
> Anyway, could you show me where it's explained that the import feature will be removed?

https://github.com/immich-app/immich/issues/4307#issuecomment-1743702907

**Reply by simulot** [reply]
> https://github.com/immich-app/immich/issues/4307#issuecomment-1743702907

The Immich server can show what's inside a folder on the server.  Images are imported in the server with the immich-CLI import command. I think this feature will be ditched at the profit of the external libraries feature. 

The "immich-go" program uploads pictures to the Immich server in the same way you'd add a photo on the web app. It saves copies of the photos in Immich folders, while the original ones are preserved.

Because the poor structure of google takeout file, I still think there isn't big advantage to import it directly. 

Immich-go read the takeout, try its best to build a coherent list of images and upload them to the server.  Then discard the takeout archive.
